### PR TITLE
[SMALLFIX] Added Sonar dependency

### PR DESCRIPTION
**Pull request by yupeng9**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -307,7 +314,7 @@
           <artifactId>maven-surefire-plugin</artifactId>
           <version>2.14</version>
           <configuration>
-            <argLine>-Djava.net.preferIPv4Stack=true -Djava.security.krb5.realm= -Djava.security.krb5.kdc=</argLine>
+            <argLine>-Djava.net.preferIPv4Stack=true -Djava.security.krb5.realm= -Djava.security.krb5.kdc= ${surefireArgLine}</argLine>
```
Comment (calvinjia) [inline pom.xml]: Extra space between `=` and `$`?

File: pom.xml [hunk 2]
```diff
@@ -476,6 +487,24 @@
             </goals>
           </execution>
         </executions>
+      </plugin>      
+      <plugin>
+        <groupId>org.jacoco</groupId>
+        <artifactId>jacoco-maven-plugin</artifactId>
+        <version>${jacoco.version}</version>
+        <executions>
+            <execution>
```
Comment (calvinjia) [inline pom.xml]: Indentation is off?

File: pom.xml [hunk 3]
```diff
@@ -326,6 +333,10 @@
             </excludes>
           </configuration>
         </plugin>
+        <plugin>
+          <groupId>org.codehous.mojo</groupId>
+          <artifacts>sonar-maven-plugin</artifacts>
```
Comment (calvinjia) [inline pom.xml]: `artifactId`?

File: pom.xml (hunk 1 above)
Comment (yupeng9) [inline pom.xml]: No, it's not part of security param

File: pom.xml (hunk 2 above)
Comment (yupeng9) [inline pom.xml]: done

**Comment by yupeng9** [comment]
Jenkins, test this please

**Pull request closed without merging**